mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -303,24 +303,34 @@ def warn_if_has_name(self, name: str) -> None:
             m = f'Overriding previous value of environment variable {name!r} with a new one'
             FeatureNew(m, '0.58.0').use(self.subproject, self.current_node)
 
+    def convert_to_path(self, value: str | ExternalProgram) -> str:
+        if isinstance(value, ExternalProgram):
+            FeatureNew('ExternalPrograms accepted by environment objects', '1.1.0').use(self.subproject, self.current_node)
+            return value.get_path()
```

Comment:
I think you should use value.get_command() that returns a list of strings. It can happen an ExternalProgram has more than one command args for example when it's a python script on Windows. Meson reads the shebang and sets `command = ['python', 'script.py']`.

The list of strings will probably have to be joined with shelex.join().

Reply:
I chose to flatten the list via itertools. Let me know if that suffices.

Reply:
I think we need to take a step back and ask, what is that supposed to accomplish? And also, what is an env object supposed to accomplish here?

Because treating it like multiple values is certainly wrong.

But also, treating it like a shell quoted value is wrong.
Unfortunately, *not* treating it like a shell quoted value is also wrong.

Reply:
I've now taken a different route; given that the method's documentation has been enhanced, I added that `full_path` must be used for environment variables too (since they take strings).